### PR TITLE
Chapter 11 account activation

### PR DESCRIPTION
**Pull request by Arizar98**

### CODE REVIEW

**Comment by Arizar98** [comment]
Ready

File: app/assets/stylesheets/account_activations.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the AccountActivations controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: https://sass-lang.com/
```
Comment (saobv-2223) [inline app/assets/stylesheets/account_activations.scss]: remove file that never used

File: app/controllers/account_activations_controller.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+class AccountActivationsController < ApplicationController
+
```
Comment (saobv-2223) [inline app/controllers/account_activations_controller.rb]: remove blank line

File: app/controllers/sessions_controller.rb [hunk 3]
```diff
@@ -5,9 +5,17 @@ def create
     user = User.find_by email: params[:session][:email].downcase
 
     if user && user.authenticate(params[:session][:password])
-      log_in user
-      params[:session][:remember_me] == "1" ? remember(user) : forget(user)
-      redirect_back_or user
+
```
Comment (saobv-2223) [inline app/controllers/sessions_controller.rb:8]: remove blank line

File: app/helpers/account_activations_helper.rb [hunk 4]
```diff
@@ -0,0 +1,2 @@
+module AccountActivationsHelper
+end
```
Comment (saobv-2223) [inline app/helpers/account_activations_helper.rb]: delete file that never used

File: app/models/user.rb [hunk 5]
```diff
@@ -32,13 +33,39 @@ def remember
...
+    # Converts email to all lower-case.
+    def downcase_email
+      self.email.downcase!
+    end
+
+    # Creates and assigns the activation token and digest.
+    def create_activation_digest
+      self.activation_token = User.new_token
+      self.activation_digest = User.digest(activation_token)
+    end
```
Comment (saobv-2223) [inline app/models/user.rb]: remove 2 spaces for those lines

**Comment by Arizar98** [comment]
Ready

**Comment by Arizar98** [comment]
Ready

File: app/controllers/account_activations_controller.rb [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+      log_in user
+      flash[:success] = "Account activated!"
+      redirect_to user
+    else
+      flash[:danger] = "Invalid activation link"
+      redirect_to root_url
+    end
+  end
+
+end
```
Comment (saobv-2223) [inline app/controllers/account_activations_controller.rb]:
```suggestion
  end
end
```

File: config/environments/development.rb [hunk 7]
```diff
@@ -1,62 +1,14 @@
...
+  config.action_mailer.delivery_method = :smtp
+  config.action_mailer.default_url_options = {host: "localhost:3000" }
+
+  config.action_mailer.smtp_settings = {
+  :user_name => '00618017908729',
+  :password => '1ef4c419cc40ec',
+  :address => 'smtp.mailtrap.io',
+  :domain => 'smtp.mailtrap.io',
+  :port => '2525',
+  :authentication => :cram_md5
```
Comment (saobv-2223) [inline config/environments/development.rb]: using gem figaro to manage env variables

**Comment by Arizar98** [comment]
Ready

**Review by saobv-2223 — COMMENTED** [review]
Your PR is having 2 commits, merge to 1 commit please

**Comment by Arizar98** [comment]
Ready

**Comment by ngocvt-0484** [comment]
merged

**Pull request merged**